cmd/commands/datasetIngestor.go

```diff
@@ -91,32 +91,33 @@ For Windows you need instead to specify -user username:password on the command l
 
 		// globus specific vars (if needed)
 		var globusClient globus.GlobusClient
-		var srcCollection, srcPrefixPath, destCollection, destPrefixPath string
-
-		switch transferType {
-		case Ssh:
-			transferFiles = cliutils.SshTransfer
-		case Globus:
-			transferFiles = cliutils.GlobusTransfer
-			var globusConfigPath string
-			if cmd.Flags().Lookup("globus-cfg").Changed {
-				globusConfigPath = globusCfgFlag
-			} else {
-				execPath, err := os.Executable()
-				if err != nil {
-					log.Fatalln("can't find executable path:", err)
+		var gConfig cliutils.GlobusConfig
+
+		if copyFlag {
```

Comment:
This doesn't quite work since we don't know if the data will actually be copied until CheckDataCentrallyAvailableSsh. Maybe it does need to be done in all cases.

I don't like how the logic about whether or not to transfer gets interleaved into all the other functions. Maybe this is something that should be refactored later.

Reply:
Sorry for the additional effort.